### PR TITLE
Bring back Get-ExchangeADPermissions

### PR DESCRIPTION
**Pull request by lusassl-msft**
**Description:**
Made some more changes to make sure that the check runs smoothly on all versions of Exchange.

### CODE REVIEW

File: Diagnostics/HealthChecker/DataCollection/ExchangeInformation/Get-ExchangeAdPermissions.ps1 [hunk 1]
```diff
@@ -99,7 +97,13 @@ Function Get-ExchangeAdPermissions {
     foreach ($domain in $forest.Domains) {
 
         $domainName = $domain.Name
-        $domainDN = $domain.GetDirectoryEntry().distinguishedName
+        try {
+            $domainDN = $domain.GetDirectoryEntry().distinguishedName
```
Comment (dpaulson45) [inline Diagnostics/HealthChecker/DataCollection/ExchangeInformation/Get-ExchangeAdPermissions.ps1:101]: why was this added? curious because we are basically bypassing this domain when it is skipped and could be vulnerable to the CVE.

File: Diagnostics/HealthChecker/DataCollection/ExchangeInformation/Get-ExchangeAdPermissions.ps1 [hunk 2]
```diff
@@ -58,13 +58,11 @@ Function Get-ExchangeAdPermissions {
 
     $groupLists = @(
         (NewGroupEntry "Exchange Trusted Subsystem" @(
-            (NewMatchingEntry -ObjectTypeGuid $altSecIdentitySchemaGUID -AdminSdHolder $false -ComputerClass $true -RootOnly $false),
-            (NewMatchingEntry -ObjectTypeGuid $altSecIdentitySchemaGUID -AdminSdHolder $true -ComputerClass $true -RootOnly $false)
+            (NewMatchingEntry -ObjectTypeGuid $altSecIdentitySchemaGUID -AdminSdHolder $false -ComputerClass $true -RootOnly $false)
```
Comment (dpaulson45) [inline Diagnostics/HealthChecker/DataCollection/ExchangeInformation/Get-ExchangeAdPermissions.ps1:61]: `RootOnly` appears to be always set to `$false` now. We should have that removed.

File: Diagnostics/HealthChecker/DataCollection/ExchangeInformation/Get-ExchangeAdPermissions.ps1 (hunk 1 above)
Comment (bill-long) [reply Diagnostics/HealthChecker/DataCollection/ExchangeInformation/Get-ExchangeAdPermissions.ps1:101]: HealthChecker is terminating in a customer environment due to unreachable domain. Check your email.

File: Diagnostics/HealthChecker/DataCollection/ExchangeInformation/Get-ExchangeAdPermissions.ps1 (hunk 2 above)
Comment (lusassl-msft) [reply Diagnostics/HealthChecker/DataCollection/ExchangeInformation/Get-ExchangeAdPermissions.ps1]: We could remove it because it doesn't play a role right now or leave it in place (if something like that is needed for future testings).

Comment (bill-long) [reply Diagnostics/HealthChecker/DataCollection/ExchangeInformation/Get-ExchangeAdPermissions.ps1]: I'd like consider a completely different API for this actually. I will push changes on a different branch.

**Review by bill-long — APPROVED** [review]

**Review by bill-long — CHANGES_REQUESTED** [review]
Waiting for E15 verification.

**Review by bill-long — APPROVED** [review]

**Review by dpaulson45 — APPROVED** [review]
